asllib/Typing.ml

```diff
@@ -560,13 +553,13 @@ module Annotate (C : ANNOTATE_CONFIG) = struct
             let+ () = check_type_satisfies' env t1 t_bool in
             let+ () = check_type_satisfies' env t2 t_bool in
             T_Bool |> with_loc
-        | AND | OR | EOR ->
+        | AND | OR | EOR (* when has_bitvector_structure env t1 ? *) ->
```

Comment:
Isn't this covered by the `check_bits_equal_widths'` just below?

The next `when has_bitvector_structure env t1` is needed because we need to differenciate it from the operation on integers.